packages/cli/src/cog/cutline.ts

```diff
@@ -136,47 +143,80 @@ export class Cutline {
+    }
+
+    /**
+     * Convert a Wsg84 polygon to it's bounds
+     */
+    private sourceWsg84PolyToBounds(poly: Position[]): Bounds {
+        const { fromWsg84 } = this.targetProj;
+        let minX = Infinity;
+        let minY = Infinity;
+        let maxX = -Infinity;
+        let maxY = -Infinity;
+        for (const p of poly) {
+            const [x, y] = fromWsg84(p);
+            if (x < minX) minX = x;
+            if (y < minY) minY = y;
+            if (x > maxX) maxX = x;
+            if (y > maxY) maxY = y;
+        }
+
+        return new Bounds(minX, minY, maxX - minX, maxY - minY);
+    }
+
     /**
      * Merge child nodes that have at least a covering fraction
      * @param quadKey the quadKey to descend
      * @param srcArea the aread of interest
      * @param minZ Don't produce any quadKeys of this length or less and creater than this length + 2
      * @param coveringFraction merge tiles that cover at least this fraction
-     * @return the quadKeys and percentage covered of the world by this srcArea set
+     * @return the quadKeys and fraction covered of the world by this srcArea set
      */
     private makeQuadKeys(
         quadKey: string,
         srcArea: MultiPolygon,
         minZ: number,
         coveringFraction: number,
     ): { quadKeys: string[]; fractionCovered: number } {
-        const clip = this.targetProj.tileToPolygon(QuadKey.toTile(quadKey));
+        const clip = this.targetProj.tileToPolygon(QuadKey.toTile(quadKey)) as Polygon;
         const intArea = intersection(srcArea, clip) as MultiPolygon | null;
-        if (intArea == null || intArea.length == 0) return { quadKeys: [], fractionCovered: 0 };
+
+        if (intArea == null || intArea.length == 0) {
+            return { quadKeys: [], fractionCovered: 0 };
+        }
         if (quadKey.length == minZ + 4) {
             return { quadKeys: [quadKey], fractionCovered: 1 };
-        } else {
-            const ans: { quadKeys: string[]; fractionCovered: number } = { quadKeys: [], fractionCovered: 0 };
-            for (let i = 0; i < 4; ++i) {
-                const { quadKeys, fractionCovered } = this.makeQuadKeys(
-                    quadKey + i.toString(),
-                    intArea,
-                    minZ,
-                    coveringFraction,
-                );
-                if (fractionCovered == 0) continue;
-                ans.fractionCovered += fractionCovered * 0.25;
-                ans.quadKeys = ans.quadKeys.concat(quadKeys);
-            }
-            if (
-                (ans.fractionCovered >= coveringFraction || quadKey.length > minZ + 2) &&
-                ans.quadKeys.length > 1 &&
-                quadKey.length >= minZ
-            )
-                ans.quadKeys = [quadKey];
-
-            return ans;
         }
+
+        const ans: { quadKeys: string[]; fractionCovered: number } = { quadKeys: [], fractionCovered: 0 };
+        for (let i = 0; i < 4; ++i) {
```

Comment:
`i` is really weird here, we also have a const of all the character names for qks.

```
for (const key of QuadKey.Keys) { ... }
```